e2e/providerconfig-aws-e2e-test/README.md

```diff
@@ -0,0 +1,190 @@
+
+### option 1. inside configuration package: make target `e2e`
+
+This make target:
+- builds the `providerconfig-aws-e2e-test` configuration package
+- spins up a local `kind` cluster
+- deploys the `providerconfig-aws-e2e-test` configuration package to the `kind` cluster
+- runs the e2e tests
+
+This make target expects the target AWS provider packages in test to be already built and pushed to
+a registry that the target EKS cluster can reach.  
+
+The make target expects the following environment variables to be set:
+
+- `AWS_FAMILY_PACKAGE_IMAGE`: The package URL for `provider-family-aws` 
+- `AWS_EC2_PACKAGE_IMAGE`: The package URL for `provider-aws-ec2`
+- `AWS_RDS_PACKAGE_IMAGE`: The package URL for `provider-aws-rds`
+- `AWS_EKS_IAM_DEFAULT_ADMIN_ROLE`: the ARN of an existing IAM role. This will be assigned as the E2E test EKS cluster default admin
+- `TARGET_CROSSPLANE_VERSION`: The target crossplane version to be deployed into the testing cluster
+- `UPTEST_CLOUD_CREDENTIALS`: The AWS credentials for the AWS account that the e2e tests will run on. Should be in the format of AWS CLI INI config. 
+
+An example usage:
+
+my-aws-creds.txt
+```ini
+[default]
+aws_access_key_id = YOUR-AWS-ACCESS-KEY
+aws_secret_access_key = your-aws-secret-access-key
+```
+
+```shell
+export AWS_FAMILY_PACKAGE_IMAGE="xpkg.upbound.io/upbound/provider-family-aws:1.4.0"
+export AWS_EC2_PACKAGE_IMAGE="xpkg.upbound.io/upbound/provider-aws-ec2:1.4.0"
+export AWS_RDS_PACKAGE_IMAGE="xpkg.upbound.io/upbound/provider-aws-rds:1.4.0"
+export AWS_EKS_IAM_DEFAULT_ADMIN_ROLE="arn:aws:iam::123456789012:role/mydefaulteksadminrole"
+export TARGET_CROSSPLANE_VERSION="1.15.2"
+export UPTEST_CLOUD_CREDENTIALS="$(cat my-aws-creds.txt)"
+# from repo root
+make -C e2e/providerconfig-aws-e2e-test e2e
+```
+
+### option 2. with provider image publish: target `providerconfig-e2e`
+
+This make target:
+- builds and publishes the providers
+- builds the `providerconfig-aws-e2e-test` configuration package
+- spins up a local `kind` cluster
+- deploys the `providerconfig-aws-e2e-test` configuration package to the `kind` cluster
+- runs the e2e tests using `uptest` with the published provider images
+
+The make target expects 
+- `XPKG_REG_ORGS`: the target OCI repository URL for provider images to be published
+- `VERSION`: the version tag of the published provider images
+- `UPTEST_CLOUD_CREDENTIALS`: The AWS credentials for the AWS account that the e2e tests will run on. Should be in the format of AWS CLI INI config.
+- `AWS_EKS_IAM_DEFAULT_ADMIN_ROLE`: the ARN of an existing IAM role. This will be assigned as the E2E test EKS cluster default admin
+
+example usage:
+```shell
+export UPTEST_CLOUD_CREDENTIALS="$(cat my-aws-creds.txt)"
+export AWS_EKS_IAM_DEFAULT_ADMIN_ROLE="arn:aws:iam::123456789012:role/mydefaulteksadminrole"
+# from repo root
+make VERSION=v1.4.0-testversion XPKG_REG_ORGS=index.docker.io/erhancag providerconfig-e2e
+```
+
+### Via Github Actions from PR 
+TBD
```

Comment:
nit: Do you think address these TBDs on this PR or future?

Reply:
This will be addressed in a separate PR. The aim for this one is to make the tests available as a make target. GH action will require some setup considerations.